test/data/resource-position.json

```diff
@@ -0,0 +1,163 @@
+            "source": {
+              "label": "anchor",
+              "type": "self"
+            },
+            "value":50
+          },
+          "timestamp": "2015-03-06T16:57:53.643+13:00",
+          "source": {
+            "label": "anchor",
+            "type": "self"
+          }
+        }
+      }
+    }
+  },
+
+  "resources": {
+    "regions":{
+        "Huge-fish-farm":{
+          "name":"Huge fish farm",
+          "description":"A great big fish farm in the middle of our bay",
+          "points":{
+            "nw":{
+              "position":{
+                "timestamp": "2015-03-06T16:57:53.643+13:00",
+                "source": {
+                  "label": "position",
+                  "type": "NMEA0183",
+                  "talker": "GP",
+                  "sentence": "RMC"
+                },
+                "longitude": 173.1693,
+                "latitude": -41.156426,
+                "altitude": 0
+              },
+              "order":1
+              },
+            "ne":{
+              "position":{
+                "timestamp": "2015-03-06T16:57:53.643+13:00",
+                "source": {
+                  "label": "position",
+                  "type": "NMEA0183",
+                  "talker": "GP",
+                  "sentence": "RMC"
+                },
+                "longitude": 173.5693,
+                "latitude": -41.156426,
+                "altitude": 0
+              },
+              "order":2
+            },
+            "south":{
+              "position":{
+                "timestamp": "2015-03-06T16:57:53.643+13:00",
+                "source": {
+                  "label": "position",
+                  "type": "NMEA0183",
+                  "talker": "GP",
+                  "sentence": "RMC"
+                },
+                "longitude": 173.1693,
+                "latitude": -41.356426,
+                "altitude": 0
+              },
+              "order":3
+            }
+          },
+          "timestamp": "2015-03-06T16:57:53.643+13:00",
+          "source": {
+            "label": "region",
+            "type": "manual"
+          }
+        }
+      },
+    "waypoints": {
+      "waypoint-1": {
+         "name": "My first waypoint",
+         "position": {
+           "timestamp": "2015-03-06T16:57:53.643+13:00",
+            "source": {
+              "label": "position",
+              "type": "NMEA0183",
+              "talker": "GP",
+              "sentence": "RMC"
+            },
+            "longitude": 173.1693,
+            "latitude": -41.156426,
+            "altitude": 0
+          },
+          "description": "",
+          "type": "fish"
+        }
+     },
+  
+    "routes":{
+      "Nelson-AbelTasman":{
+        "name":"Nelson to Abel Tasman Park",
+        "description": "From Nelson Cut to just off Fishermans Island",
+        "distance":23.4,
+        "waypoints":[
+          {
+            "waypoint-1":{
+              "order": 0
```

Comment:
I don't think your argument holds water. It's like adding line numbers to a text file so that it doesn't spontaneously reorder. What if some software turns your precious order properties to strings and orders the waypoints lexically?